src/shillelagh/backends/apsw/vt.py

```diff
@@ -547,8 +583,8 @@ def Filter(  # pylint: disable=too-many-locals
             kwargs["limit"] = limit
         if self.adapter.supports_offset:
             kwargs["offset"] = offset
-        if self.adapter.supports_requested_columns:
-            kwargs["requested_columns"] = self.requested_columns
+        if self.adapter.supports_requested_columns and len(index) == 3:
```

Comment:
If you use object top level in JSON, the self.adapter.supports_requested_columns check becomes unneeded